src/main/java/payapal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package main.java.payapal;
+
+public class Perfection {
+
+    private static Perfection perf;
+
+    public synchronized static Perfection getPerf() {
+        if (perf == null) {
+            perf = new Perfection();
+        }
+        return perf;
+    }
+
+
+    public static boolean isPerfect(long candidate) {
+        boolean retVal;
+        long[] divisors = GetDivisors(candidate);
+        int sum = 0;
+        for (int d = 0; d < 1000; d++)
+        {
+            sum = sum + divisors[d];
+        }
+        if (sum == candidate)
+            retVal = true;
+        return retVal;
+    }
+
+
+    private static long[] GetDivisors(long candidate) {
+        long[] divisors = new long[];
+        int d = 0;
+        for (long i = 0; i < candidate; i++) {
+            long foo = candidate / i;
+            if (foo * i == candidate) {
```

Comment:
This condition will always return true, as `candidate` is always equal to `candidate / i * i` (the / i * i cancel each other), causing the wrong divisors to be added into the divisors array. Instead, the correct condition is `candidate % i == 0`. In this way, the right divisors will be get. Also, the `long foo = candidate / i;` should be removed.